marker_driver_rustc/src/conversion/marker/pat.rs

```diff
@@ -0,0 +1,113 @@
+                    if index >= offset_pos {
+                        index += offset_pos;
+                    }
+                    StructFieldPat::new(
+                        self.to_span_id(pat.span),
+                        self.to_symbol_id_for_num(u32::try_from(index).expect("a index over 2^32 us unexpected")),
+                        self.to_pat(pat),
+                    )
+                }));
+                PatKind::Struct(
+                    self.alloc(|| StructPat::new(data, self.to_path_from_qpath(qpath), api_fields, ddpos.is_some())),
+                )
+            },
+            hir::PatKind::Or(pats) => PatKind::Or(
+                self.alloc(|| OrPat::new(data, self.alloc_slice_iter(pats.iter().map(|rpat| self.to_pat(rpat))))),
+            ),
+            hir::PatKind::Tuple(pats, dotdot) => {
+                let pats = if let Some(rest_pos) = dotdot.as_opt_usize() {
+                    let (start, end) = pats.split_at(rest_pos);
+                    self.chain_pats(start, self.new_rest_pat(), end)
+                } else {
+                    self.alloc_slice_iter(pats.iter().map(|pat| self.to_pat(pat)))
+                };
+                PatKind::Tuple(self.alloc(|| TuplePat::new(data, pats)))
+            },
+            hir::PatKind::Box(_) => PatKind::Unstable(self.alloc(|| UnstablePat::new(data))),
+            hir::PatKind::Ref(pat, muta) => {
+                PatKind::Ref(self.alloc(|| RefPat::new(data, self.to_pat(pat), matches!(muta, hir::Mutability::Mut))))
+            },
+            hir::PatKind::Slice(start, wild, end) => {
+                let elements = if let Some(wild) = wild {
+                    self.chain_pats(start, self.to_pat(wild), end)
+                } else {
+                    assert!(end.is_empty());
+                    self.alloc_slice_iter(start.iter().map(|pat| self.to_pat(pat)))
+                };
+                PatKind::Slice(self.alloc(|| SlicePat::new(data, elements)))
+            },
+            // These haven't been implemented yet, as they require expressions.
+            // The pattern creation is tracked in #50
+            hir::PatKind::Path(_) => todo!("{pat:#?}"),
+            hir::PatKind::Lit(_) => todo!("{pat:#?}"),
+            hir::PatKind::Range(_, _, _) => todo!("{pat:#?}"),
+        }
+    }
+
+    fn chain_pats(
+        &self,
+        start: &[hir::Pat<'tcx>],
+        ast_wild: PatKind<'ast>,
+        end: &[hir::Pat<'tcx>],
+    ) -> &'ast [PatKind<'ast>] {
+        let start = start.iter().map(|pat| self.to_pat(pat));
+        let middle = std::iter::once(ast_wild);
+        let end = end.iter().map(|pat| self.to_pat(pat));
+        let api_pats: Vec<_> = start.chain(middle).chain(end).collect();
+        self.alloc_slice_iter(api_pats.into_iter())
+    }
+
+    fn new_rest_pat(&self) -> PatKind<'ast> {
+        // This is a dummy span, it's dirty, but at least works for the mean time :)
+        let span = rustc_span::Span::default();
```

Comment:
I think it would be clearer to use [`DUMMY_SP`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_span/constant.DUMMY_SP.html), but maybe we should have a constant like this in marker itself so there would be no need to make a conversion on the next line.

Reply:
I'd say that using a dummy span is actually a bad smell. This case is intended to be temporary until, we have the first expression backend and can properly construct the span. It's annoying, that rustc doesn't store the span of `..`.

I'll create an issue to fix this, once this PR is merged :)